### ISSUE TITLE
Expected drag drop context

### ISSUE THREAD

**Issue by mindell** [strapi/strapi#22392]
## Bug report

```
Invariant Violation: Expected drag drop context
    at invariant (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-4B74B645.js?v=f8369899:66:15)
    at useDragDropManager (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-LK544M46.js?v=f8369899:176:3)
    at useDragLayer (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-LK544M46.js?v=f8369899:182:27)
    at DragLayer (http://localhost:1337/node_modules/.strapi/vite/deps/layout-CzSSEy9b-ONCEVVWO.js?v=f8369899:107:85)
    at renderWithHooks (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-IYHTQODQ.js?v=f8369899:11568:26)
    at mountIndeterminateComponent (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-IYHTQODQ.js?v=f8369899:14946:21)
    at beginWork (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-IYHTQODQ.js?v=f8369899:15934:22)
    at beginWork$1 (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-IYHTQODQ.js?v=f8369899:19781:22)
    at performUnitOfWork (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-IYHTQODQ.js?v=f8369899:19226:20)
    at workLoopSync (http://localhost:1337/node_modules/.strapi/vite/deps/chunk-IYHTQODQ.js?v=f8369899:19165:13)
```
      
### Required System information

Environment: development
OS: linux-x64
Strapi Version: 5.5.0
Node/Yarn Version: npm/10.9.2 node/v20.11.0 linux x64 workspaces/false
Edition: Community
Database: postgres

- Is your project Javascript or Typescript: Typescript

### Describe the bug
The error only appears when opening the Content Manager. Other pages have no problem.

### Steps to reproduce the behavior

1. Go to Content Manager
2. See error

### Expected behavior
I should be able to see the Content Manager without any problems.

### Code snippets

If applicable, add code samples to help explain your problem.

**Comment by Kyle772** [comment]
+1

**Comment by MarkusAbtion** [comment]
This is a real bug with a reproducible root cause, not an invalid report — adding a diagnosis since both this and #22792 were auto-closed by the template bot without one. A fix is open at #27217.

**Root cause:** `react-dnd` holds its `DndContext` at module scope, so the admin bundle must contain exactly one copy of it. Two packages each declare it as their own direct dependency:

- `@strapi/admin` → `react-dnd: "16.0.1"` — renders the shared `<DndProvider>` in `AuthenticatedLayout`
- `@strapi/content-manager` → `react-dnd: "16.0.1"` — calls `useDragLayer` in its layout

npm's default hoisting collapses those onto a single physical copy, so provider and consumer share a context — but that is a property of the install layout, not of the code. Any tree that keeps the copies separate gives them two different `DndContext` objects, `useDragDropManager` reads the one `DndProvider` never populated, gets `undefined`, and throws. That is why it takes out the Content Manager specifically while the rest of the admin keeps working.

Triggers include pnpm's strict `node_modules` isolation, npm `install-strategy=nested`, a plugin or app pinning its own `react-dnd`, and `overrides`/`resolutions` that split the version.

`react-dnd` is absent from Strapi's `ADMIN_VITE_DEDUPE_MODULES`, the list that already force-dedupes `react-redux` (added for the same failure in `could not find react-redux context value` form) and the CodeMirror packages (same failure in `instanceof` form). It was never added because hoisting normally hides it.

**Measured**, same source tree both times, only the install strategy differing — counting react-dnd's own invariant string in the built admin bundle is a direct count of bundled copies:

| install | physical `react-dnd` dirs | invariant string in bundle | Content Manager |
| --- | --- | --- | --- |
| `install-strategy=nested` | 4 | 4 | crashes |
| default (hoisted) | 1 | 1 | works |

**Workaround until #27217 lands** — get the tree down to one physical copy:

```bash
find node_modules -type d -name react-dnd   # expect exactly one
```

If there is more than one, remove whatever is preventing dedupe (the nested/strict install setting, or a conflicting pin), delete `node_modules` **and** the lockfile, reinstall so the tree is re-resolved rather than rebuilt from the old shape, then clear the Vite dep cache at `<app>/node_modules/.strapi/vite` and rebuild. Re-resolving from scratch is the part I verified; `npm dedupe` on an already-fragmented tree may be enough but I have not tested it.

Note that upgrading Strapi does not help — the fragmentation comes from the install layout, not from Strapi's version.

Worth highlighting from your report: it shows `workspaces/false`, so npm workspaces are not required to reproduce this — any non-deduped tree does it.
